Sim/SimG4Full/src/lib/ParticleHistoryAction.cpp

```diff
@@ -0,0 +1,31 @@
+#include "SimG4Full/ParticleHistoryAction.h"
+
+#include "SimG4Common/EventInformation.h"
+
+#include "G4EventManager.hh"
+#include "G4LorentzVector.hh"
+
+namespace sim {
+
+ParticleHistoryAction::ParticleHistoryAction() {}
+
+void ParticleHistoryAction::PreUserTrackingAction(const G4Track* aTrack) {}
+
+void ParticleHistoryAction::PostUserTrackingAction(const G4Track* aTrack) {
+  auto g4EvtMgr = G4EventManager::GetEventManager();
+  auto evtinfo = dynamic_cast<sim::EventInformation*>(g4EvtMgr->GetUserInformation());
+  G4LorentzVector prodPos(aTrack->GetGlobalTime() - aTrack->GetLocalTime(), aTrack->GetVertexPosition());
+  G4LorentzVector endPos(aTrack->GetGlobalTime(), aTrack->GetPosition());
+  if (selectSecondary(*aTrack)) {
+    evtinfo->addParticle(aTrack);
+  }
+}
+
+bool ParticleHistoryAction::selectSecondary(const G4Track& aTrack) {
+  G4LorentzVector p4(aTrack.GetMomentum(), aTrack.GetTotalEnergy());
+  if (p4.e() < 5.0) {  // this value is very arbitrary
```

Comment:
This is rather intransparent. It would be preferable if the user can configure the cuts and furthermore define different criteria for different regions. E.g. in the calorimeter there will be different selection criteria than in the tracker.

Reply:
Hi Julia, thanks for your comments. You won't be able to tell secondaries from primaries by looking at the status, as this property is used by the event generator and all the particles will be "final state" particles by the time they reach the detector simulation. However you can check if they have a startvertex. I nevertheless set the status to "1" for completeness. 

I made the energy Cut is configurable now - as we discussed in the meeting there are different usecases for this tool, and the current implementation should cover the usecase  for the tracker alone. I propose to implement regions and support the calorimeter in a future pr ( if there is interest ).

Reply:
Only for the secondaries a startvertex is set. (The primaries do actually have a production vertex in Geant 4, I'm unsure myself if it is not better to set it).